app/src/main/java/dev/ricsarabia/cryptochallenge/ui/main/BooksAdapter.kt

```diff
@@ -0,0 +1,32 @@
+package dev.ricsarabia.cryptochallenge.ui.main
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import dev.ricsarabia.cryptochallenge.databinding.BookItemBinding
+import dev.ricsarabia.cryptochallenge.domain.Book
+
+class BooksAdapter(val onBookClick: (Book)->Unit): RecyclerView.Adapter<BooksViewHolder>() {
+    var books = listOf<Book>()
+        set(value) { field = value; notifyDataSetChanged() }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BooksViewHolder {
+        val binding = BookItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return BooksViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: BooksViewHolder, position: Int) {
+        holder.bind(books[position], onBookClick)
+    }
+
+    override fun getItemCount() = books.size
+}
+
+class BooksViewHolder(val binding: BookItemBinding) : RecyclerView.ViewHolder(binding.root) {
+    fun bind(book: Book, onBookClick: (Book)->Unit) {
```

Comment:
What do you think about to use "with(binding)" to improve your code? :D 
```suggestion
    fun bind(book: Book, onBookClick: (Book)->Unit) = with(binding) {
         root.setOnClickListener { onBookClick(book) }
         bookNameTextView.text = book.book
         Glide.with(root).load(book.imageUrl).into(bookImageView)
    }
```